hw/drivers/sensors/lis2dh12/src/lis2dh12.c

```diff
@@ -640,13 +640,20 @@ int
 lis2dh12_pull_up_disc(struct sensor_itf *itf, uint8_t disconnect)
 {
     uint8_t reg;
+    int rc;
 
-    reg = 0;
+    rc = lis2dh12_readlen(itf, LIS2DH12_REG_CTRL_REG0, &reg, 1);
+    if (rc) {
```

Comment:
@benmccrea I think it would be better to make the read of the bit conditional given it is not applicable for lis2dh12.

Reply:
@vrahane Thanks, I did it this way to make it work for both lis2dk10 and lis2dh12. If you want, I can add a syscfg and use the old behavior unless the syscfg is set.

Reply:
Yes, I think that would be better given the register is reserved for the lis2dk10

Reply:
@vrahane I added LIS2DH12_LIS2DK10_COMPAT. Let me know if you have any other concerns or can approve.